opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1854,6 +1854,66 @@ const std::map<std::string,int>& CpGrid::getLgrNameToLevel() const{
     return lgr_names_;
 }
 
+bool CpGrid::isOnLgrBoundary(int faceIdx) const
+{
+    const auto& faceToCell = current_view_data_->face_to_cell_[cpgrid::EntityRep<1>(faceIdx, true)];
+    const auto& cell1 = Dune::cpgrid::Entity<0>(*current_view_data_,faceToCell[0].index(), true);
+    if (faceToCell.size() == 1) { // face is on the boundary of the grid
+        assert(cell1.level() > 0);
+        return cell1.level();
+    }
+    else {
+        const auto& cell2 = Dune::cpgrid::Entity<0>(*current_view_data_,faceToCell[1].index(), true);
+        return ( (cell1.level() == 0) && (cell2.level() > 0) ) || ( (cell1.level() > 0) && (cell2.level() == 0) );
+    }
+}
+
+const cpgrid::EntityRep<1> CpGrid::getParentFaceFromLgrBoundaryFace(int faceOnBoundaryLgrIdx) const
```

Comment:
I need to think about this code. It uses a lot of implementation details. Hopefully we can do something similar using the DUNE grid interface.